scalyr_agent/scalyr_monitor.py

```diff
@@ -212,6 +212,13 @@ def __init__(
 
         StoppableThread.__init__(self, name="metric thread")
 
+    @property
+    def stop_agent_on_failure(self):
```

Comment:
Should we also have a method of setting this programmatically via code?

Imagine a scenario where some monitors manager / other codes decides, based on some criteria that monitor should be treated as essential.

I know we don't really have good patterns establishes for that yet - e.g. hoes does "code vs config" precedence works, but we do have some use cases like that already in the wild, notably Kubernetes Explorer and related monitors - those monitors should only be be treated as essential in case a specific environment is set.

I guess technically this could be achieved by overriding this method on this ScalyrMonitor class, so it's probably fine.

Reply:
I guess we can at least introduce instance attribute so we can manipulate it during initialization.